test/unit/test-amp-img.js

```diff
@@ -665,4 +666,111 @@ describe('amp-img', () => {
       );
     });
   });
+  describe('layout intrinsic', () => {
+    let browser;
+    beforeEach(() => {
+      fixture.iframe.height = 800;
+      fixture.iframe.width = 800;
+      browser = new BrowserController(fixture.win);
+    });
+    it('should not exceed given width and height even if image\
+      natural size is larger', () => {
+      let ampImg;
+      return getImg({
+        src: '/examples/img/sample.jpg', // 641 x 481
+        width: 100,
+        height: 100,
+        layout: 'intrinsic',
+      })
+        .then(image => {
+          ampImg = image;
+          return browser.waitForElementLayout('amp-img');
```

Comment:
This wait was explicitly added to accommodate Firefox. `addElement` actually calls and waits for build and layout callbacks explicitly, and `amp-img`'s layoutCallback actually waits for the load promise, which only returns after the `load` event. I'm wondering what are the guarantees for `load` event. Are we guaranteed at this point in time that the image has laid out properly? Are we guaranteed at this point that images are properly sized? 

The failure here in FF (repros on MacOS) is that getBoundingClientRect is returning 0x0 dimensions for the `<img>` inside the `<amp-img>`.

Reply:
Okay, well FF on Windows still doesn't like this implementation because its `getBoundingClientRect` returns 0x0, whereas its `clientHeight` and `clientWidth`, `offsetHeight` and `offsetWidth` values are all correct. If we proceed to test `client` or `offset` values however, Safari isn't happy. I'm going to skip FF for now and add a note. I've manually verified that this isn't actually broken on FF Windows and it's only a test flake. I'll leave the which-measurements-are-happy-on-which-browsers question to a future date.